### PR TITLE
insert automatic line breaks in legends for oversized legend entries in time profile plots

### PR DESCRIPTION
**Pull request by sfrechen**
@Yuri05

### CODE REVIEW

File: src/code/qualification/addLegendWithLineBreaks.m [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        if size(str(1:II(end-ii)),2) <= NoCharPerLine
+            str1 = str(1:II(end-ii));
+            str2 = str((II(end-ii)+1):end);
+            res = sprintf('%s%s%s',str1,'\n',str2);
+            break
+        end
+    end
+else
+    res = str;
+end
```
Comment (Yuri05) [inline src/code/qualification/addLegendWithLineBreaks.m:48]: * if str contains neither `" "` nor `"-"`, this function will return empty string. 
* the function will insert max. 1 linebreak. If the length of str is e.g. > 2*NoCharPerLine - the second line will be still too long

File: src/code/qualification/plotQualificationTimeProfile.m [hunk 2]
```diff
@@ -83,7 +83,7 @@ function plotQualificationTimeProfile(WSettings,plotIndex,TimeProfile,ObservedDa
     curvesLegend=[curvesLegend legLabel];
     curvesHandle=[curvesHandle p_handle];
 end
-legend(curvesHandle, curvesLegend);
+addLegendWithLineBreaks(gcf,curvesHandle,curvesLegend);
```
Comment (Yuri05) [inline src/code/qualification/plotQualificationTimeProfile.m:86]: only for time profile? Other plot types do not have the "too long legend"-problem?

**Review by Yuri05 — CHANGES_REQUESTED** [review]
s. comments

File: src/code/qualification/addLegendWithLineBreaks.m (hunk 1 above)
Comment (sfrechen) [reply src/code/qualification/addLegendWithLineBreaks.m:48]: @Yuri05 Fixed with commit c3eed69

File: src/code/qualification/plotQualificationTimeProfile.m (hunk 2 above)
Comment (sfrechen) [reply src/code/qualification/plotQualificationTimeProfile.m:86]: @Yuri05 yes, I am not aware of issues elsewhere.

File: src/code/qualification/addLegendWithLineBreaks.m (hunk 1 above)
Comment (sfrechen) [reply src/code/qualification/addLegendWithLineBreaks.m:48]: Example:
image

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**